packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -138,7 +146,11 @@ Future<int> runCommandAndStreamOutput(
   RegExp filter,
   StringConverter mapFunction,
   Map<String, String> environment,
+  RegExp detachFilter,
+  Future<void> Function(Process) onDetach,
+  Future<void> Function(Process) onExit,
 }) async {
+  final Completer<int> result = Completer<int>();
   final Process process = await runCommand(
```

Comment:
In the detach case, should the process be started with `ProcessStartMode.detachedWithStdio` ?

Reply:
Detached mode creates a separate session, but that doesn't seem to have any influence on `ios-deploy`.

Reply:
Got it. I've added `detachedWithStdio`.